src/main/java/uk/gov/justice/digital/delius/transformers/OffenderTransformer.java

```diff
@@ -315,5 +304,169 @@ private Human humanOf(String forename, String forename2, String surname) {
+                .isCurrentRo(isCurrentRo(om.getResponsibleOfficer()))
+                .isCurrentOm(isCurrentManager(om.getActiveFlag(), om.getEndDate()))
+                .isCurrentPom(false)
+                .omStartDate(localDateOf(om.getAllocationDate()))
+                .omEndDate(localDateOf(om.getEndDate()))
+                .build();
+
+        return mo;
+    }
+
+    private ManagedOffender convertToManagedOffender(uk.gov.justice.digital.delius.jpa.standard.entity.PrisonOffenderManager pom, Staff staff) {
+
+         ManagedOffender mo = uk.gov.justice.digital.delius.data.api.ManagedOffender.builder()
+                .staffCode(staff.getOfficerCode())
+                .offenderId(pom.getOffenderId())
+                .nomsNumber(Optional.ofNullable(pom.getManagedOffender()).map(o -> o.getNomsNumber()).orElse(null))
+                .crnNumber(Optional.ofNullable(pom.getManagedOffender()).map(o -> o.getCrn()).orElse(null))
+                .offenderSurname(Optional.ofNullable(pom.getManagedOffender()).map(o -> o.getSurname()).orElse(null))
+                .isCurrentRo(isCurrentRo(pom.getResponsibleOfficer()))
+                .isCurrentOm(false)
+                .isCurrentPom(isCurrentManager(pom.getActiveFlag(),pom.getEndDate()))
+                .omStartDate(null)
+                .omEndDate(null)
+                .build();
+
+        return mo;
+    }
+
+    public List<uk.gov.justice.digital.delius.data.api.ManagedOffender> managedOffenderOf(Staff staff, boolean current) {
+
+         List<ManagedOffender> managedOffenders = new ArrayList<>();
+
+        if (staff.getOffenderManagers() != null && !staff.getOffenderManagers().isEmpty()) {
+            managedOffenders.addAll(
+                    staff.getOffenderManagers().stream()
+                            .map(offenderManager -> convertToManagedOffender(offenderManager, staff))
+                            .collect(Collectors.toList()));
+        }
+
+        if (staff.getPrisonOffenderManagers() != null && !staff.getPrisonOffenderManagers().isEmpty()) {
+            managedOffenders.addAll(
+                    staff.getPrisonOffenderManagers().stream()
+                    .map(prisonOffenderManager -> convertToManagedOffender(prisonOffenderManager, staff))
+                    .collect(Collectors.toList()));
+        }
+
+        if (current && !managedOffenders.isEmpty()) {
+
+            // Filter the list of managed offenders to only those which hold currently active OM, POM or RO roles
+            return managedOffenders
+                    .stream()
+                    .filter(mo -> (mo.isCurrentOm() || mo.isCurrentPom() || mo.isCurrentRo()))
+                    .collect(Collectors.toList());
+        }
+
+        return managedOffenders;
+    }
+
+
+    private boolean isCurrentRo(uk.gov.justice.digital.delius.jpa.standard.entity.ResponsibleOfficer ro) {
+        boolean result = false;
```

Comment:
Minor: Maybe we could do `return Optional.ofNullable(ro).filter(not(Responsible:Officer::getEndDate)).isPresent()` for readability

Reply:
I will take a look again and see which reads best. Thanks.